core/Cargo.toml

```diff
@@ -27,6 +28,9 @@ thiserror = "1.0.60"
 
 [dev-dependencies]
 athena-vm = { path = "../vm/entrypoint" }
+athena-vm-sdk = { path = "../vm/sdk" }
```

Comment:
I'm not sure, is this the right direction of dependencies that core depends on the SDK? As this is only needed for the tests, how about defining the needed things in the test code instead? All it takes to get rid of it is:

```
diff --git a/core/src/runtime/mod.rs b/core/src/runtime/mod.rs
index 0aafae3..decdf23 100644
--- a/core/src/runtime/mod.rs
+++ b/core/src/runtime/mod.rs
@@ -809,7 +809,6 @@ pub mod tests {
     ADDRESS_LENGTH, SOME_COINS,
   };
   use athena_vm::helpers::address_to_32bit_words;
-  use athena_vm_sdk::{Pubkey, PUBKEY_LENGTH};
   use wallet_common::SendArguments;

   use crate::{
@@ -869,7 +868,7 @@ pub mod tests {
     let program = wallet_program();

     // set up host context
-    let owner_pubkey = [1u8; PUBKEY_LENGTH];
+    let owner_pubkey = [1u8; 32];
     let principal = ADDRESS_ALICE;
     let template = ADDRESS_BOB;
     let callee = ADDRESS_CHARLIE;
@@ -901,7 +900,7 @@ pub mod tests {
     #[allow(clippy::arc_with_non_send_sync)]
     let mut runtime = Runtime::new(program, Some(host.clone()), opts, Some(ctx));
     let mut stdin = AthenaStdin::new();
-    stdin.write(&Pubkey(owner_pubkey));
+    stdin.write(&owner_pubkey);
     runtime.write_vecs(&stdin.buffer);

     // make sure the program loaded correctly
```

Reply:
It somehow bothers me to have the same type defined multiple times in multiple places. Maybe this also belongs in `athena-interface`.

Reply:
It's 1 constant defining the length of the public key. Btw, is the public key length part of the VM? Isn't it _host_ who defines what public keys are and it's "opaque" for the VM?